### ISSUE TITLE
require destructured from `expect` from test context, when using snapshots in concurrent tests

### ISSUE THREAD

**Issue by dimaMachina** [veritem/eslint-plugin-vitest#49]
❌ Fail
```js
it("test", () => {
  expect(1).toMatchSnapshot()
})
```

✅ Pass
```js
it("test", ({ expect }) => {
  expect(1).toMatchSnapshot()
})
```

helped fix `Error: Snapshot cannot be used outside of test` error https://github.com/vitest-dev/vitest/issues/2729#issuecomment-1399470875

**Comment by veritem** [comment]
This should be a different rule that handles this kind of behavior. Open for pull requests

**Comment by connorjs** [comment]
Should there be an option to enforce this only for `concurrent`, but allow it to be forced for everything?

Note: I’m new to Vitest. From my reading of the docs, I though the destructured form was only “needed” in the concurrent case.

**Comment by veritem** [comment]
Yeah, destructuring is needed for `concurrent` tests. Wouldn't this be great if it was a different rule?

**Comment by connorjs** [comment]
Yes, I think it should be a new rule.

- I think it would be included in the `recommended` set to require it for `concurrent` cases.
- Assuming you can restructure in the non-concurrent case (I'm new to Vitest), I think we should have an option similar to `always` that enforces it everywhere. (This would make it easier to switch to concurrent.)
- Ideally, this would be auto-fixable.

I have never written an ESLint rule, so I do not know how “involved” this would be to submit a PR, but it could be good learning.

**Comment by veritem** [comment]
this was resolved check [docs](https://github.com/veritem/eslint-plugin-vitest/blob/main/docs/rules/require-local-test-context-for-concurrent-snapshots.md)
